src/Widgets/TerminalWidget.vala

```diff
@@ -776,21 +776,27 @@ namespace Terminal {
                     var uris = selection_data.get_uris ();
                     string path;
                     File file;
-
                     for (var i = 0; i < uris.length; i++) {
-                         file = File.new_for_uri (uris[i]);
-                         if ((path = file.get_path ()) != null) {
-                             uris[i] = Shell.quote (path) + " ";
+                        // Get sanitized unquoted path (Files sends uris that are escaped and quoted)
```

Comment:
> Files sends uris that are escaped and quoted

I'm not sure if this behavior is just about Files. I'm not sure about each implementation, but this branch works as expected if I open Terminal from Code, GNOME Files, and Dolphin, the file manager made for KDE.

(I'm concerning this comment sounds like as if this code only supports URIs passed from Files and does not support them passed from other apps)

Reply:
I have amended the comment accordingly